components/canvas/webgl_thread.rs

```diff
@@ -200,13 +236,23 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
     }
 
     /// Handles an unlock external callback received from webrender::ExternalImageHandler
-    fn handle_unlock(&mut self, context_id: WebGLContextId) {
-        let data = Self::make_current_if_needed(context_id, &self.contexts, &mut self.bound_context_id)
-                        .expect("WebGLContext not found in a WebGLMsg::Unlock message");
-        let info = self.cached_context_info.get_mut(&context_id).unwrap();
-        if let Some(gl_sync) = info.gl_sync.take() {
-            // Release the GLSync object.
-            data.ctx.gl().delete_sync(gl_sync);
+    fn handle_unlock(&mut self, context_id: WebGLContextId, webgl_chan: &WebGLChan) {
+        {
+            let data = Self::make_current_if_needed(context_id, &self.contexts, &mut self.bound_context_id)
+                            .expect("WebGLContext not found in a WebGLMsg::Unlock message");
+            let info = self.cached_context_info.get_mut(&context_id).unwrap();
```

Comment:
nit: while you're touching this, using the indexing operator may give a nicer error message if it fails.